stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/ImageLoader.kt

```diff
@@ -38,6 +42,29 @@ public object ImageLoader {
             }
     }
 
+    public suspend fun getBitmapUri(
+        context: Context,
+        url: String,
+        transformation: ImageTransformation = ImageTransformation.None
+    ): Uri? = getBitmap(context, url, transformation)?.getLocalBitmapUri(context)
+
+    private fun Bitmap.getLocalBitmapUri(context: Context): Uri? {
+        var bmpUri: Uri? = null
+        try {
+            val file = File(
+                context.getExternalFilesDir(Environment.DIRECTORY_PICTURES) ?: context.cacheDir,
+                "share_image_" + System.currentTimeMillis() + ".png"
+            )
+            val out = FileOutputStream(file)
+            compress(Bitmap.CompressFormat.PNG, 90, out)
+            out.close()
```

Comment:
This will cause resource leaks. If the code crashes before `out.close` gets called the resource will stay open and memory won't be free. You can use `FileOutputStream(file).use` or use a `try/catch/finally` structure and call `close()` inside the finally block.

Reply:
+1 for `use` here